parent/pom.xml

```diff
@@ -2103,6 +2133,71 @@
                </plugin>
 
                <plugin>
+                  <groupId>org.codehaus.mojo</groupId>
+                  <artifactId>build-helper-maven-plugin</artifactId>
+                  <version>1.7</version>
+                  <executions>
+                     <execution>
+                        <id>add-source</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                           <goal>add-source</goal>
+                        </goals>
+                        <configuration>
+                           <sources>
+                              <source>${dir.ispn}/atomic-factory/src/main/java</source>
+                              <source>${dir.ispn}/cdi/common/src/main/java</source>
+                              <source>${dir.ispn}/cdi/embedded/src/main/java</source>
+                              <source>${dir.ispn}/cdi/remote/src/main/java</source>
+                              <source>${dir.ispn}/checkstyle/src/main/java</source>
+                              <source>${dir.ispn}/cli/cli-client/src/main/java</source>
+                              <source>${dir.ispn}/cli/cli-interpreter/src/main/java</source>
+                              <source>${dir.ispn}/client/hotrod-client/src/main/java</source>
```

Comment:
@slaskawi I have wrote about that in the comment above while sending pull request: 

In the pom.xml you can see the big list of src folders which is added to sources of maven. This is done for jacoco, so that the generated report can also show the covered and uncovered lines directly on the class files.

The build-helper-maven-plugin does not support the src folders inclusion via wildcards or include/exclude tags. If this is ok, I will file an issue in plugins repo, and as soon as it is implemented will change my code correspondingly.

If the new module is created and the dev forgot to add it here, it will not cause any problem. The result would be only that the generated jacoco report will show the coverage for the corresponding package on method level and not directly on the source code. 

You can see it in the comment attachment: the Report_with_source1.png and Report_with_source2.png files show the generated jacoco report for specified class when the source for that module was available above.; the Report_with_missing_source.png shows the generated report for the same class in case when the source for that class is not found during jacoco profile run (as you can see the method names are not clickable so you cannot view the coverage directly in the code, to see what is covered and what is not).
![report_with_missing_source](https://cloud.githubusercontent.com/assets/1311306/13813154/3cc485ee-eb80-11e5-934c-2d0a6f095265.png)
![report_with_source1](https://cloud.githubusercontent.com/assets/1311306/13813153/3cc4405c-eb80-11e5-9751-3b96ae30f538.png)
![report_with_source2](https://cloud.githubusercontent.com/assets/1311306/13813152/3cc29090-eb80-11e5-93ba-d495aa3b08f2.png)